modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterLayerResponse.java

```diff
@@ -804,6 +804,9 @@ private MosaicElement createMosaic() throws IOException {
                                     // Adding globalRoi to the output
                                     RenderedOp rop = (RenderedOp) mosaic;
                                     rop.setProperty("ROI", in.roi);
+
+                                    mosaic = footprintBehavior.postProcessMosaic(mosaic, in.roi,
```

Comment:
Hmm, maybe, yeah. I guess in a case where your raster fills the entire image bounds and your footprint has, say, a hole right in the middle then it should need to do the footprint handling as well right? I'll try a test case and see what happens.